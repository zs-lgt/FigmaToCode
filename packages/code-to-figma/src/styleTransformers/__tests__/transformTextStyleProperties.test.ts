import { transformTextStyleProperties } from '../transformTextStyleProperties';

describe('transformTextStyleProperties', () => {
    it('textAlign auto', () => {
        const result = transformTextStyleProperties({ textAlign: 'auto' });
        expect(result).toMatchSnapshot();
    });

    it('textAlign center', () => {
        const result = transformTextStyleProperties({ textAlign: 'center' });
        expect(result).toMatchSnapshot();
    });

    it('textAlign justify', () => {
        const result = transformTextStyleProperties({ textAlign: 'justify' });
        expect(result).toMatchSnapshot();
    });

    it('lineHeight "50%"', () => {
        const result = transformTextStyleProperties({ lineHeight: '50%' });
        expect(result).toMatchSnapshot();
    });

    it('lineHeight 1.4', () => {
        const result = transformTextStyleProperties({ lineHeight: 14 });
        expect(result).toMatchSnapshot();
    });

    it('lineHeight auto', () => {
        const result = transformTextStyleProperties({ lineHeight: 'auto' });
        expect(result).toMatchSnapshot();
    });

    it('letterSpacing "50%"', () => {
        const result = transformTextStyleProperties({ letterSpacing: '50%' });
        expect(result).toMatchSnapshot();
    });

    it('letterSpacing 1.4', () => {
        const result = transformTextStyleProperties({ letterSpacing: 1.4 });
        expect(result).toMatchSnapshot();
    });

    it('textDecorationLine line-through', () => {
        const result = transformTextStyleProperties({ textDecorationLine: 'line-through' });
        expect(result).toMatchSnapshot();
    });

    it('text shadow', () => {
        const result = transformTextStyleProperties({
            textShadowColor: '#000000',
            textShadowRadius: 15,
            textShadowOffset: {
                width: 10,
                height: 20
            }
        });
        expect(result).toMatchSnapshot();
    });

    it('Non standard fontStyle warning', () => {
        const warn = jest.fn();
        // @ts-ignore
        global.console = { warn };
        transformTextStyleProperties({
            fontStyle: 'solid'
        });
        expect(warn).toHaveBeenCalledTimes(1);
        expect(warn).toHaveBeenCalledWith(
            "fontStyle: 'solid': Non-standard font styles may not work at other platforms"
        );
    });

    it("fontStyle normal doesn't affected warning", () => {
        const warn = jest.fn();
        // @ts-ignore
        global.console = { warn };
        transformTextStyleProperties({
            fontStyle: 'normal'
        });
        expect(warn).toHaveBeenCalledTimes(0);
    });

    it("undefined fontStyle doesn't affected warning", () => {
        const warn = jest.fn();
        // @ts-ignore
        global.console = { warn };
        transformTextStyleProperties({});
        expect(warn).toHaveBeenCalledTimes(0);
    });
});
