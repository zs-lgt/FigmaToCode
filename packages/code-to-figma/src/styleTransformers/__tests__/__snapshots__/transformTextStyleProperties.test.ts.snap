// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`transformTextStyleProperties letterSpacing "50%" 1`] = `
Object {
  "fills": Array [
    Object {
      "color": Object {
        "b": 0,
        "g": 0,
        "r": 0,
      },
      "opacity": 1,
      "type": "SOLID",
    },
  ],
  "fontName": Object {
    "family": "Inter",
    "style": "Regular",
  },
  "letterSpacing": Object {
    "unit": "PERCENT",
    "value": 50,
  },
}
`;

exports[`transformTextStyleProperties letterSpacing 1.4 1`] = `
Object {
  "fills": Array [
    Object {
      "color": Object {
        "b": 0,
        "g": 0,
        "r": 0,
      },
      "opacity": 1,
      "type": "SOLID",
    },
  ],
  "fontName": Object {
    "family": "Inter",
    "style": "Regular",
  },
  "letterSpacing": Object {
    "unit": "PIXELS",
    "value": 1.4,
  },
}
`;

exports[`transformTextStyleProperties lineHeight "50%" 1`] = `
Object {
  "fills": Array [
    Object {
      "color": Object {
        "b": 0,
        "g": 0,
        "r": 0,
      },
      "opacity": 1,
      "type": "SOLID",
    },
  ],
  "fontName": Object {
    "family": "Inter",
    "style": "Regular",
  },
  "lineHeight": Object {
    "unit": "PERCENT",
    "value": 50,
  },
}
`;

exports[`transformTextStyleProperties lineHeight 1.4 1`] = `
Object {
  "fills": Array [
    Object {
      "color": Object {
        "b": 0,
        "g": 0,
        "r": 0,
      },
      "opacity": 1,
      "type": "SOLID",
    },
  ],
  "fontName": Object {
    "family": "Inter",
    "style": "Regular",
  },
  "lineHeight": Object {
    "unit": "PIXELS",
    "value": 14,
  },
}
`;

exports[`transformTextStyleProperties lineHeight auto 1`] = `
Object {
  "fills": Array [
    Object {
      "color": Object {
        "b": 0,
        "g": 0,
        "r": 0,
      },
      "opacity": 1,
      "type": "SOLID",
    },
  ],
  "fontName": Object {
    "family": "Inter",
    "style": "Regular",
  },
  "lineHeight": Object {
    "unit": "AUTO",
  },
}
`;

exports[`transformTextStyleProperties text shadow 1`] = `
Object {
  "effects": Array [
    Object {
      "blendMode": "NORMAL",
      "color": Object {
        "a": 1,
        "b": 0,
        "g": 0,
        "r": 0,
      },
      "offset": Object {
        "x": 10,
        "y": 20,
      },
      "radius": 15,
      "spread": 0,
      "type": "DROP_SHADOW",
      "visible": true,
    },
  ],
  "fills": Array [
    Object {
      "color": Object {
        "b": 0,
        "g": 0,
        "r": 0,
      },
      "opacity": 1,
      "type": "SOLID",
    },
  ],
  "fontName": Object {
    "family": "Inter",
    "style": "Regular",
  },
}
`;

exports[`transformTextStyleProperties textAlign auto 1`] = `
Object {
  "fills": Array [
    Object {
      "color": Object {
        "b": 0,
        "g": 0,
        "r": 0,
      },
      "opacity": 1,
      "type": "SOLID",
    },
  ],
  "fontName": Object {
    "family": "Inter",
    "style": "Regular",
  },
}
`;

exports[`transformTextStyleProperties textAlign center 1`] = `
Object {
  "fills": Array [
    Object {
      "color": Object {
        "b": 0,
        "g": 0,
        "r": 0,
      },
      "opacity": 1,
      "type": "SOLID",
    },
  ],
  "fontName": Object {
    "family": "Inter",
    "style": "Regular",
  },
  "textAlignHorizontal": "CENTER",
}
`;

exports[`transformTextStyleProperties textAlign justify 1`] = `
Object {
  "fills": Array [
    Object {
      "color": Object {
        "b": 0,
        "g": 0,
        "r": 0,
      },
      "opacity": 1,
      "type": "SOLID",
    },
  ],
  "fontName": Object {
    "family": "Inter",
    "style": "Regular",
  },
  "textAlignHorizontal": "JUSTIFIED",
}
`;

exports[`transformTextStyleProperties textDecorationLine line-through 1`] = `
Object {
  "fills": Array [
    Object {
      "color": Object {
        "b": 0,
        "g": 0,
        "r": 0,
      },
      "opacity": 1,
      "type": "SOLID",
    },
  ],
  "fontName": Object {
    "family": "Inter",
    "style": "Regular",
  },
  "textDecorationLine": "STRIKETHROUGH",
}
`;
